internal/app/cf-terraforming/cmd/generate.go

```diff
@@ -688,6 +688,57 @@ func generateResources() func(cmd *cobra.Command, args []string) {
 					}
 				}
 			}
+
+		case "cloudflare_pages_domain":
```

Comment:
i a bit confused here. why do we fetch all the pages projects and then iterate all of them to get pages domains?

ideally, these methods should fetch all the results from a single endpoint and use that to dump the output. we shouldn't need to make additional calls or build other structs in order to generate a single resource output.

Reply:
This is needed because there is no endpoint to list all Pages Domains. The endpoint to list pages domains (https://api.cloudflare.com/#pages-domains-get-domains) requires a parameter of the project name, so I need to list all projects to then go through and get all domains.

Reply:
Hi! I just started using this tool and this is a missing feature I'd like to have. Is it possible to set the domain as a variable, instead of looking for every domain?

Reply:
The purpose of generating is to get all the possible resources for output.

As an FYI, I haven't worked on in this in a while as I've run into issues with the pages project part.

Reply:
I was thinking it could be alike to the DNS resources: I have to call it for each zone separately.

So if it worked like the DNS resources, I just would call it per domain.

Reply:
The API doesn't work that way. DNS records are zone level which are seperated. Pages are account level resources, so they include all domains for all zones in the account.